zebra-state/src/service/finalized_state.rs

```diff
@@ -321,6 +347,77 @@ impl FinalizedState {
         result
     }
 
+    #[cfg(feature = "elasticsearch")]
+    /// Store finalized blocks into an elasticsearch database.
+    ///
+    /// We use the elasticsearch bulk api to index multiple blocks at a time while we are
+    /// synchronizing the chain, when we get close to tip we index blocks one by one.
+    pub fn elasticsearch(
+        &mut self,
+        block: &Arc<block::Block>,
+        committed_tip_height: &Option<block::Height>,
+    ) {
+        if let Some(client) = self.elastic_db.clone() {
+            let block_time = block.header.time.timestamp();
+            let local_time = chrono::Utc::now().timestamp();
+
+            const AWAY_FROM_TIP_BULK_SIZE: usize = 800;
+            const CLOSE_TO_TIP_BULK_SIZE: usize = 2;
+            const CLOSE_TO_TIP_SECONDS: i64 = 14400; // 4 hours
+
+            // If we are close to the tip index one block per bulk call.
+            let mut blocks_size_to_dump = AWAY_FROM_TIP_BULK_SIZE;
+            if local_time - block_time < CLOSE_TO_TIP_SECONDS {
+                blocks_size_to_dump = CLOSE_TO_TIP_BULK_SIZE;
+            }
+
+            // Insert the operation line.
+            let height_number = committed_tip_height.unwrap_or(block::Height(0)).0;
+            self.elastic_blocks.push(
+                serde_json::json!({
+                    "index": {
+                        "_id": height_number.to_string().as_str()
```

Comment:
Should the index be different every time? This is currently the finalized tip height before the block is committed, is it supposed to be the block's height?

Reply:
Yea, `_id` needs to be different for each block or it will overwrite the data. Height is what makes more sense but you are right, it should be the block height from the block. I fixed this at https://github.com/ZcashFoundation/zebra/pull/6274/commits/8b2c474aa4432c19e374269c310573d897cbb8a9